### ISSUE TITLE
Resource Not Found when editing customer account

### ISSUE THREAD

**Issue by teamcrisis** [mailchimp/mc-magento2#344]
### Preconditions
1. Magento 2.2.5
2. Mailchimp 1.0.29

### Steps to reproduce

1. Do not set Mailchimp Store/ Mailchimp List details in the Mailchimp General Configuration
2. Edit customer account in magento admin. Customers > All customers > edit

### Actual and Expected result

Expected result:
    To be able to edit the customer account, without error, even if Mailchimp settings are not correct to connect to the API.

Actual result:
    Error `Resource Not Found : The requested resource could not be found.` and all customer data is missing from the page.

**Comment by teamcrisis** [comment]
This appears to be related: https://github.com/mailchimp/mc-magento2/issues/240

**Comment by gonzaloebiz** [comment]
Hi @teamcrisis 

please follow the [issue reporting guidelines](https://github.com/mailchimp/mc-magento2/wiki/Issue-reporting-guidelines)
Once you meet the requirements we will re open the issue

Best

**Comment by teamcrisis** [comment]
done

**Comment by gonzaloebiz** [comment]
Can you please upload an image of your configuration page?

**Comment by teamcrisis** [comment]
![mailchimp](https://user-images.githubusercontent.com/4693056/42397465-170583fc-8134-11e8-9d87-a5e8154a64c3.png)

**Comment by gonzaloebiz** [comment]
But.. this configuration not meet your preconditions :(

>Steps to reproduce
>Do not set Mailchimp Store/ Mailchimp List details in the Mailchimp General Configuration

**Comment by teamcrisis** [comment]
Correct. For whatever reason, when I initially tried to setup the Mailchimp API information, it would not fetch my Store/List info from Mailchimp. 

I stopped working on it for about two weeks and have now returned to the issue and find that it can now connect to mailchimp and retrieve the store/list info, despite me not changing anything in the config. Therefore, I set the store/list info to what they need to be and the issue is resolved.

However, that does not change the fact that a mailchimp API connection failure or not setting the store/list details will cause magento functionality to break. 

It seems at least one other issue https://github.com/mailchimp/mc-magento2/issues/328 could be related to API connection problems.

**Comment by gonzaloebiz** [comment]
If you enable the extension in the configuration page but you don't finish the configuration (p.e. if you don't select a MailChimp store), this is a wrong configuration and not an issue of the extension.

**Comment by teamcrisis** [comment]
As stated, I attempted to finish the configuration, but the extension would not connect to mailchimp despite me providing valid keys. I returned to the issue some time later and the connection problem seemed to have resolved itself. I don't know, maybe there was a mailchimp API issue at the time I was initially trying to set it up. It still doesn't change the fact that if a connection cannot be made, magento functionality breaks. Checks should be in place to bypass the extension if an API failure occurs and allow the remaining functionality to work. As an admin, I should be able to edit a customer account whether or not the mailchimp extension is setup correctly or if an API connection can be established. A failure in the mailchimp extension should not cause the entire customer account to become unusable - it should simply not allow data syncing to occur.
